File: Bitwise Operation/136. Single Number/main.go
package main

import (
	"fmt"
)

func singleNumber(nums []int) int {
	ret := 0
	for _, num := range nums {
		ret = ret ^ num
	}
	return ret
}

func main() {
	nums := []int{3, 1, 2, 2, 1}
	fmt.Println("Output: ", singleNumber(nums))
}

File: Bitwise Operation/136. Single Number/main.c
#include <stdio.h>
#include <stdlib.h>

int singleNumber(int* nums, int numsSize){
    int result = 0;
    int i = 0;
    for (i = 0; i < numsSize; i++) {
        result = result ^ nums[i];
    }
    return result;
}

int main() {
    int nums[] = {3, 1, 2, 2, 1};
    printf("Output: %d.", singleNumber(nums, 5));
    return 0;
}